sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/SequencePointCollector.java

```diff
@@ -59,9 +59,17 @@ private static void publishCoverage(Coverage coverage, String filePath, List<Seq
         int line = lineSequencePoints.getKey();
 
         List<SequencePoint> linePoints = lineSequencePoints.getValue();
-        int coveredPoints = (int)linePoints.stream().filter(point -> point.getHits() > 0).count();
 
-        coverage.addBranchCoverage(filePath, new BranchCoverage(line, linePoints.size(), coveredPoints));
+        Map<String, List<SequencePoint>> groupsByCoverageKey = linePoints.stream().collect(Collectors.groupingBy(SequencePoint::getCoverageKey));
```

Comment:
I would a comment above `// the key is the SequencePoint unique identifier, the value is a list of SequencePoint information coming from different sources (e.g. different coverage reports for the same file)`

I cannot think of a better name at this point

Reply:
maybe `uniqueSequencePointInfo`